dpctl/tensor/_searchsorted.py

```diff
@@ -0,0 +1,172 @@
+            "Unrecognized value of 'side' keyword argument. "
+            "Expected either 'left' or 'right'"
+        )
+
+    if sorter is None:
+        q = du.get_execution_queue([x1.sycl_queue, x2.sycl_queue])
+    else:
+        q = du.get_execution_queue(
+            [x1.sycl_queue, x2.sycl_queue, sorter.sycl_queue]
+        )
+    if q is None:
+        raise du.ExecutionPlacementError(
+            "Execution placement can not be unambiguously "
+            "inferred from input arguments."
+        )
+
+    if x1.ndim != 1:
+        raise ValueError("First argument array must be one-dimensional")
+
+    x1_dt = x1.dtype
+    x2_dt = x2.dtype
+
+    host_evs = []
+    ev = dpctl.SyclEvent()
+    if sorter is not None:
+        if not isdtype(sorter.dtype, "integral"):
+            raise ValueError(
+                f"Sorter array must have integral data type, got {sorter.dtype}"
+            )
+        if x1.shape != sorter.shape:
+            raise ValueError(
+                "Sorter array must be one-dimension with the same "
+                "shape as the first argument array"
+            )
+        res = empty(x1.shape, dtype=x1_dt, usm_type=x1.usm_type, sycl_queue=q)
+        ind = (sorter,)
+        axis = 0
+        wrap_out_of_bound_indices_mode = 0
+        ht_ev, ev = ti_take(
+            x1,
+            ind,
+            res,
+            axis,
+            wrap_out_of_bound_indices_mode,
+            sycl_queue=q,
+            depends=[
+                ev,
+            ],
+        )
+        x1 = res
+        host_evs.append(ht_ev)
+
+    if x1_dt != x2_dt:
+        dt = result_type(x1, x2)
+        if x1_dt != dt:
+            x1_buf = _empty_like_orderK(x1, dt)
+            ht_ev, ev = ti_copy(
+                src=x1,
+                dst=x1_buf,
+                sycl_queue=q,
+                depends=[
+                    ev,
+                ],
+            )
+            host_evs.append(ht_ev)
+            x1 = x1_buf
+        if x2_dt != dt:
+            x2_buf = _empty_like_orderK(x2, dt)
+            ht_ev, ev = ti_copy(
+                src=x2,
+                dst=x2_buf,
+                sycl_queue=q,
+                depends=[
+                    ev,
+                ],
+            )
+            host_evs.append(ht_ev)
+            x2 = x2_buf
+
+    dst_usm_type = du.get_coerced_usm_type([x1.usm_type, x2.usm_type])
+    dst_dt = int32 if x2.size <= iinfo(int32).max else int64
+
+    dst = _empty_like_orderK(x2, dst_dt, usm_type=dst_usm_type)
+
+    if side == "left":
+        ht_ev, _ = _searchsorted_left(
+            hay=x1,
+            needles=x2,
+            positions=dst,
+            sycl_queue=q,
+            depends=[
+                ev,
+            ],
+        )
+    else:
+        ht_ev, _ = _searchsorted_right(
+            hay=x1,
+            needles=x2,
+            positions=dst,
+            sycl_queue=q,
+            depends=[
+                ev,
+            ],
+        )
```

Comment:
I'm not sure what's better here, but something to consider is that `side` could be passed to a `searchsorted` Python API, and the internals of the lambda could pick between the `_searchsorted_left` and `_searchsorted_right` tables when calling.

This approach worked well in `repeat` and kept the Python calls a little cleaner.

Reply:
Fair point. I will try this out in a subsequent PR.